app/src/main/java/com/github/fribourgsdp/radio/GoogleSignInActivity.kt

```diff
@@ -0,0 +1,140 @@
+package com.github.fribourgsdp.radio
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import com.github.fribourgsdp.radio.databinding.ActivityGoogleSignInBinding
+import com.google.android.gms.auth.api.signin.GoogleSignIn
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount
+import com.google.android.gms.auth.api.signin.GoogleSignInClient
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.AuthCredential
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import java.lang.Exception
+
+
+class GoogleSignInActivity : AppCompatActivity() {
+
+
+    //view binding
+    private lateinit var binding: ActivityGoogleSignInBinding
+
+    private lateinit var googleSignInClient: GoogleSignInClient
+    private lateinit var firebaseAuth: FirebaseAuth
+
+    //constants
+    private companion object{
+        private const val TAG = "GOOGLE_SIGN_TAG"
+    }
+
+
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityGoogleSignInBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        //configure for the Google SignTn
+        val googleSignInOptions = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
```

Comment:
This reference is not found in `string.xml`. Am I missing something ?

Reply:
The reference is in a values.xml file generated automatically during the build under app/build/generated/res/google-services/release/values but the android studio interface doesn't always find it. But when you build the app, this is never a problem and the reference is found.